sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -113,7 +113,8 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
       planToCache: LogicalPlan,
       tableName: Option[String],
       storageLevel: StorageLevel): Unit = {
-    if (lookupCachedData(planToCache).nonEmpty) {
+    if (storageLevel == StorageLevel.NONE) {
```

Comment:
Maybe just put a comment in the body indicating that this is intentional and why